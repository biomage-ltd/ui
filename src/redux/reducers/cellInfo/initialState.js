const initialState = {
  focus: {
    store: 'cellSets',
    key: 'louvain',
  },
  groupedTrack: 'louvain',
  selectedTracks: ['louvain'],
};

export default initialState;
