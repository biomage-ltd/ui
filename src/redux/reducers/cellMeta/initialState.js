const mitochondrialContent = {
  loading: true,
  error: false,
  data: [],
};

const doubletScores = {
  loading: true,
  error: false,
  data: [],
};

const initialState = {
  mitochondrialContent,
  doubletScores,
};

export default initialState;
