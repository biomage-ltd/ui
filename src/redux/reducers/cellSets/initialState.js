const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
