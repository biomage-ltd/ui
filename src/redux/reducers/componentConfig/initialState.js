import {
  legendBaseState,
  dimensionsBaseState,
  axesBaseState,
  titleBaseState,
  fontStyleBaseState,
  colourBaseState,
  markerBaseState,
  labelBaseState,
} from './baseStylesState';

// PLOTS & TABLES - Categorical Embedding
const embeddingCategoricalInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// PLOTS & TABLES - Continuous Embedding
const embeddingContinuousInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  logEquation: 'datum.expression*1',
  shownGene: 'notSelected',
  selectedSample: 'All',
};

// PLOTS & TABLES - Heatmap
const heatmapInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    show: true,
    position: 'horizontal',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 500,
    height: 500,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedGenes: [],
  selectedCellSet: 'louvain',
  labelColour: 'transparent',
};

// PLOTS & TABLES - Volcano plot
const volcanoInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: dimensionsBaseState,
  marker: {
    ...markerBaseState,
    size: 32,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log2 fold change',
    yAxisText: ' - log10(adj.p - value)',
    gridOpacity: 5,
    offset: 10,
  },
  title: titleBaseState,
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  label: labelBaseState,
  noDifferenceColor: '#aaaaaa',
  significantUpregulatedColor: '#0000ffaa',
  significantDownregulatedColor: '#ff0000',
  notSignificantDownregulatedColor: '#aaaaaa',
  notSignificantUpregulatedColor: '#aaaaaa',
  significantChangeDirectionUnknownColor: '#aaaaaa',

  // `null` automatically scales to range. This is a problem
  // because our DE is bad right now, so it throws off the
  // range to extreme values. TODO: set this back when we have
  // good DE
  // logFoldChangeDomain: null,

  logFoldChangeDomain: 20,
  maxNegativeLogpValueDomain: null,
  negLogpValueThreshold: 4,
  logFoldChangeThreshold: 1,
  logFoldChangeTickCount: 5,
  negativeLogpValueTickCount: 5,
  downsampleRatio: 0,
  showLogFoldChangeThresholdGuides: true,
  showpvalueThresholdGuides: true,
  thresholdGuideWidth: 1,
  logFoldChangeThresholdColor: '#ff0000',
  pvalueThresholdColor: '#ff0000',
  textThresholdValue: 240,
  strokeOpa: 1,
  strokeCol: '#000000',
};

// PLOTS & TABLES - Frequency
const frequencyInitialConfig = {
  spec: '1.0.0',
  frequencyType: 'proportional',
  proportionGrouping: '',
  legend: {
    ...legendBaseState,
    title: 'Cell Set',
    position: 'top',
    offset: 40,
  },
  label: labelBaseState,
  dimensions: dimensionsBaseState,
  marker: markerBaseState,
  colour: colourBaseState,
  title: titleBaseState,
  axes: {
    ...axesBaseState,
    xAxisText: 'Sample',
    yAxisText: 'Proportion',
    offset: 10,
  },
  fontStyle: fontStyleBaseState,
  xAxisGrouping: '',
  axisTitlesize: 13,
  geneexpLegendloc: '',
};

// EMBEDDING PREVIEW - Coloured by sample
const embeddingPreviewBySampleInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Coloured by cell sets
const embeddingPreviewByCellSetsInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Config for fraction of mitochondrial reads
const embeddingPreviewMitochondrialContentInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedSample: 'All',
};

// EMBEDDING PREVIEW - Config for doublet score
const embeddingPreviewDoubletScoreInitialConfig = {
  spec: '1.0.0',
  legend: legendBaseState,
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    dx: 0,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedSample: 'All',
};

const interactiveHeatmapInitialConfig = {
  selectedTracks: ['louvain'],
  groupedTracks: ['sample', 'louvain'],
  expressionValue: 'raw',
  legendIsVisible: true,
};

// CELL SIZE DISTRIBUTION - Cell Size Distribution Histogram
const cellSizeDistributionHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top-left',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: '#UMIs in cell',
    yAxisText: '#UMIs * #Cells',
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  minCellSize: 10800,
  binStep: 200,
};

// CELL SIZE DISTRIBUTION - Cell Size Distribution Knee Plot
const cellSizeDistributionKneePlot = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top-left',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Cell rank',
    yAxisText: '#UMIs in cell',
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  minCellSize: 990,
};

// MITOCHONDRIAL CONTENT - Mitochondrial Fraction Histogram
const mitochondrialFractionHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top-right',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Fraction of mitochondrial reads',
    yAxisText: 'Fraction of cells',
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  binStep: 0.05,
  maxFraction: 0.1,
};

// MITOCHONDRIAL CONTENT - Mitochondrial Fraction Log Histogram
const mitochondrialFractionLogHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top-right',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10(#UMIs in cell)',
    yAxisText: 'Fraction of mitochondrial reads',
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
    dx: 0,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  binStep: 0.05,
  maxFraction: 3.5,
};

// CLASSIFIER - Classifier Empty Drops
const classifierEmptyDropsPlot = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    enabled: false,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 630,
    height: 500,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10[ cell size (UMIs) ]',
    yAxisText: 'classifier prob',
    gridOpacity: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  minProbability: 0.82,
  bandwidth: -1,
};

// GENES VS UMIS - Features vs UMIs
const featuresVsUMIsHistogram = {
  spec: '1.0.0',
  dimensions: {
    ...dimensionsBaseState,
    width: 630,
    height: 500,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10 [molecules]',
    yAxisText: 'Frequency',
    gridOpacity: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  label: labelBaseState,
  lowCutoff: 4.8,
  upCutoff: 2.1,
  binStep: 0.05,
};

// GENES VS UMIS - Features vs UMIs
const featuresVsUMIsScatterplot = {
  spec: '1.0.0',
  dimensions: {
    ...dimensionsBaseState,
    width: 630,
    height: 500,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'log10 [molecule counts]',
    yAxisText: 'log10 [gene counts]',
    gridOpacity: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  label: labelBaseState,
  lowCutoff: 4.8,
  upCutoff: 2.1,
};

// DOUBLE FILTER - Doublet filter histogram
const doubletScoreHistogram = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top-right',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 530,
    height: 400,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'Probability of being a doublet',
    yAxisText: 'Frequency',
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  binStep: 0.05,
  probThreshold: 0.2,
};

// DATA INTEGRATION - Embedding by Samples
const dataIntegrationEmbeddingInitialConfig = {
  spec: '1.0.0',
  legend: {
    ...legendBaseState,
    position: 'top',
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  axes: {
    ...axesBaseState,
    xAxisText: 'UMAP 1',
    yAxisText: 'UMAP 2',
    offset: 10,
  },
  title: {
    ...titleBaseState,
    fontSize: 20,
  },
  fontStyle: fontStyleBaseState,
  colour: colourBaseState,
  marker: markerBaseState,
  label: labelBaseState,
  selectedCellSet: 'louvain',
  selectedSample: 'All',
};

// DATA INTEGRATION - Frequency
const dataIntegrationFrequencyInitialConfig = {
  ...frequencyInitialConfig,
  xAxisGrouping: 'louvain',
  proportionGrouping: 'sample',
  legend: {
    ...legendBaseState,
    title: 'Sample',
    position: 'top',
    offset: 10,
  },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },

  axes: {
    ...axesBaseState,
    xAxisText: 'Louvain clusters',
    yAxisText: 'Proportion',
    offset: 10,
  },
  fontStyle: fontStyleBaseState,
};

// DATA INTEGRATION - Elbow
const dataIntegrationElbowPlotInitialConfig = {
  legend: {
    ...legendBaseState,
    position: 'top',
  },
  label: { ...labelBaseState },
  dimensions: {
    ...dimensionsBaseState,
    width: 700,
    height: 550,
  },
  marker: { ...markerBaseState },
  fontStyle: fontStyleBaseState,
  axes: {
    ...axesBaseState,
    xAxisText: 'Principal Components',
    yAxisText: 'Proportion of Variance Explained',
    titleFont: 'sans-serif',
    labelFont: 'sans-serif',
    labelFontSize: 13,
    offset: 0,
    gridOpacity: 10,
  },
  colour: {
    toggleInvert: '#FFFFFF',
  },
  title: {
    ...titleBaseState,
    font: 'sans-serif',
    fontSize: 12,
  },
  signals: [
    {
      name: 'interpolate',
      value: 'linear',
      bind: {
        input: 'select',
        options: [
          'basis',
          'cardinal',
          'catmull-rom',
          'linear',
          'monotone',
          'natural',
          'step',
          'step-after',
          'step-before',
        ],
      },
    },
  ],
};

const initialPlotConfigStates = {
  cellSizeDistributionHistogram,
  cellSizeDistributionKneePlot,
  mitochondrialFractionHistogram,
  mitochondrialFractionLogHistogram,
  classifierEmptyDropsPlot,
  featuresVsUMIsHistogram,
  featuresVsUMIsScatterplot,
  doubletScoreHistogram,
  embeddingCategorical: embeddingCategoricalInitialConfig,
  embeddingContinuous: embeddingContinuousInitialConfig,
  heatmap: heatmapInitialConfig,
  volcano: volcanoInitialConfig,
  frequency: frequencyInitialConfig,
  embeddingPreviewBySample: embeddingPreviewBySampleInitialConfig,
  embeddingPreviewByCellSets: embeddingPreviewByCellSetsInitialConfig,
  embeddingPreviewMitochondrialContent: embeddingPreviewMitochondrialContentInitialConfig,
  embeddingPreviewDoubletScore: embeddingPreviewDoubletScoreInitialConfig,
  dataIntegrationEmbedding: dataIntegrationEmbeddingInitialConfig,
  dataIntegrationFrequency: dataIntegrationFrequencyInitialConfig,
  dataIntegrationElbow: dataIntegrationElbowPlotInitialConfig,
};

const initialComponentConfigStates = {
  interactiveHeatmap: interactiveHeatmapInitialConfig,
};
export { initialPlotConfigStates, initialComponentConfigStates };

const initialState = {};
export default initialState;
