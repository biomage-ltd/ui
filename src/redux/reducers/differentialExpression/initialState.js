const initialState = {
  properties: {
    data: [],
    cellSets: {},
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'between',
    group: {
      cellSet: null,
      compareWith: null,
      basis: null,
    },
  },
};

export default initialState;
