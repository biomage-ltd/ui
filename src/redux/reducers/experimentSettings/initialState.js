const initialPipelineState = {
  startDate: null,
  endDate: null,
  status: null,
  completedSteps: [],
};

const initialWorkerState = {
  status: null,
  started: false,
  ready: null,
  restartCount: 0,
};

const initialState = {
  pipelineStatus: {
    loading: false,
    error: false,
    status: {
    },
  },
  info: {
    experimentId: null,
    experimentName: null,
    projectId: null,
    createdAt: null,
    lastViewed: null,
  },
  processing: {
    meta: {
      loading: true,
      completingStepError: false,
      loadingSettingsError: false,
      selectedConfigureEmbeddingPlot: 'sample',
    },
  },
};

export default initialState;
export { initialPipelineState, initialWorkerState };
