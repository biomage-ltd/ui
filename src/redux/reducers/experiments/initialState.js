const experimentTemplate = {
  projectUuid: null,
  name: null,
  description: 'Analysis description',
  id: null,
  createdDate: null,
  lastViewed: null,
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
  },
  ids: [],
};

export default initialState;
export { experimentTemplate };
