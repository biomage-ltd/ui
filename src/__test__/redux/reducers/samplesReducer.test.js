import samplesReducer from '../../../redux/reducers/samples';
import initialState, { sampleTemplate, sampleFileTemplate } from '../../../redux/reducers/samples/initialState';

import {
  SAMPLES_CREATE,
  SAMPLES_UPDATE,
  SAMPLES_FILE_UPDATE,
  SAMPLES_LOADED,
  SAMPLES_ERROR,
  SAMPLES_DELETE,
} from '../../../redux/actionTypes/samples';

describe('samplesReducer', () => {
  const mockUuid1 = 'asd123';
  const mockUuid2 = 'qwe234';
  const fileName = 'features.tsv';

  const sample1 = {
    ...sampleTemplate,
    name: 'test sample',
    uuid: mockUuid1,
    createdDate: '2021-01-01T14:48:00.000Z',
    lastModified: '2021-01-01T14:48:00.000Z',
  };

  const sample2 = {
    ...sampleTemplate,
    name: 'test sample 2',
    uuid: mockUuid2,
    createdDate: '2021-01-02T14:48:00.000Z',
    lastModified: '2021-01-02T14:48:00.000Z',
  };

  const updateActionResult = {
    ...sample1,
    name: 'updated name',
  };

  const oneSampleState = {
    ...initialState,
    ids: [...initialState.ids, sample1.uuid],
    [sample1.uuid]: sample1,
  };

  const twoSamplesState = {
    ...oneSampleState,
    ids: [...oneSampleState.ids, sample2.uuid],
    [sample2.uuid]: sample2,
  };

  const mockFile = {
    ...sampleFileTemplate,
    name: fileName,
  };

  it('Reduces identical state on unknown action', () => expect(
    samplesReducer(undefined, {
      action: 'well/this/is/not/a/valid/action',
      payload: {},
    }),
  ).toEqual(initialState));

  it('Inserts a new sample correctly', () => {
    const newState = samplesReducer(initialState, {
      type: SAMPLES_CREATE,
      payload: {
        sample: sample1,
      },
    });

    expect(newState.ids).toEqual([sample1.uuid]);
    expect(newState[sample1.uuid]).toEqual(sample1);
    expect(newState).toMatchSnapshot();
  });

  it('Adds a new sample correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_CREATE,
      payload: {
        sample: sample2,
      },
    });

    expect(newState.ids).toEqual([sample1.uuid, sample2.uuid]);
    expect(newState[sample1.uuid]).toEqual(sample1);
    expect(newState[sample2.uuid]).toEqual(sample2);
    expect(newState).toMatchSnapshot();
  });

  it('Updates a sample correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        sample: updateActionResult,
      },
    });

    expect(newState.ids).toEqual(oneSampleState.ids);
    expect(newState[sample1.uuid]).toEqual(updateActionResult);
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample files correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_FILE_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        file: mockFile,
        lastModified: 'newLastModified',
      },
    });

    expect(newState[sample1.uuid].fileNames).toEqual(new Set(fileName));
    expect(newState[sample1.uuid].files[fileName]).toEqual(mockFile);
    expect(newState).toMatchSnapshot();
  });

  it('Delete samples correctly', () => {
    const newState = samplesReducer(twoSamplesState, {
      type: SAMPLES_DELETE,
      payload: {
        sampleUuids: [sample2.uuid],
      },
    });

    expect(newState.ids).toEqual([sample1.uuid]);
    expect(newState[sample2.uuid]).toBeUndefined();
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample files correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_FILE_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        file: mockFile,
        lastModified: 'newLastModified',
      },
    });

    expect(newState[sample1.uuid].fileNames).toEqual(new Set(fileName));
    expect(newState[sample1.uuid].files[fileName]).toEqual(mockFile);
  });

  it('Loads samples correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_LOADED,
      payload: {
        samples: {
          ids: [sample1.uuid, sample2.uuid],
          [sample1.uuid]: sample1,
          [sample2.uuid]: sample2,
        },
      },
    });

    expect(newState.ids).toEqual([mockUuid1, mockUuid2]);
    expect(newState.meta.loading).toEqual(false);
    expect(newState.meta.error).toEqual(false);
  });

  it('Handles errors correctly', () => {
    const error = 'Failed uploading samples';

    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_ERROR,
      payload: {
        error,
      },
    });

    expect(newState.meta.loading).toEqual(false);
    expect(newState.meta.error).toEqual(error);
  });
});
