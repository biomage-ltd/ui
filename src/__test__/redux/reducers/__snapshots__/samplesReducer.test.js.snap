// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samplesReducer Adds a new sample correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
    "qwe234",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
  "qwe234": Object {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample 2",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Delete samples correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Deletes sample metadata correctly 1`] = `
Object {
  "[object Object]": Object {
    "metadata": Object {
      "metadata-test": "old-value",
    },
  },
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Inserts a new sample correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Inserts sample metadata correctly 1`] = `
Object {
  "[object Object]": Object {
    "metadata": Object {},
  },
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {
      "metadata-test": "value",
    },
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Loads samples correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
    "qwe234",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
  "qwe234": Object {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample 2",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Sets up saved state correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`samplesReducer Sets up saving state correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": true,
  },
}
`;

exports[`samplesReducer Stores error state correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": "Error message",
    "loading": false,
    "saving": false,
  },
}
`;

exports[`samplesReducer Updates a sample correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {},
    "name": "updated name",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Updates sample files correctly 1`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [
      "features.tsv",
    ],
    "files": Object {
      "features.tsv": Object {
        "error": false,
        "mime": "",
        "name": "features.tsv",
        "objectKey": "",
        "path": "",
        "size": 0,
        "success": false,
      },
    },
    "lastModified": "newLastModified",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Updates sample files correctly 2`] = `
Object {
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [
      "features.tsv",
    ],
    "files": Object {
      "features.tsv": Object {
        "error": false,
        "mime": "",
        "name": "features.tsv",
        "objectKey": "",
        "path": "",
        "size": 0,
        "success": false,
      },
    },
    "lastModified": "newLastModified",
    "metadata": Object {},
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;

exports[`samplesReducer Updates sample metadata correctly 1`] = `
Object {
  "[object Object]": Object {
    "metadata": Object {
      "metadata-test": "old-value",
    },
  },
  "asd123": Object {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": Object {
      "metadata-test": "new-value",
    },
    "name": "test sample",
    "projectUuid": null,
    "species": null,
    "toJSON": [Function],
    "type": null,
    "uuid": "asd123",
  },
  "ids": Array [
    "asd123",
  ],
  "meta": Object {
    "error": false,
    "loading": true,
    "saving": false,
  },
}
`;
